src/sagemaker/model.py

```diff
@@ -407,6 +408,19 @@ def _compilation_job_config(
             else input_shape,
             "Framework": framework.upper(),
         }
+        if framework.upper() == "PYTORCH" and framework_version is not None:
+            if target_instance_type is not None:
+                if target_instance_type.startswith("ml_"):
+                    if "ml_inf" not in target_instance_type:
+                        input_model_config["FrameworkVersion"] = utils.get_short_version(
+                            framework_version
+                        )
+                    else:
+                        raise ValueError(
```

Comment:
This seems to exclude deployment to ml_inf as well. Is that accurate?

Reply:
Yes, when the model will not be compiled against this target, then there will be no deploy for this.